### PR TITLE
Add avoid_flush_during_shutdown DB option

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Summary:
Add avoid_flush_during_shutdown DB option.

Test Plan:
See the new unit test.

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
cc @benrnd

File: HISTORY.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # Rocksdb Change Log
+## Unreleased
+### Public API Change
+
+### New Features
+* Add avoid_flush_during_shutdown option, which speeds up DB shutdown by not flushing unpersisted data. Use it only when data loss is acceptable. The options is dynamically changeable.
```
Comment (ajkr) [inline HISTORY.md]: if WAL is enabled, shouldn't there be no data loss?

**Comment by yiwu-arbug** [comment]
Comment updated. @ajkr any more comments?

**Comment by yiwu-arbug** [comment]
Rebase.

**Comment by yiwu-arbug** [comment]
ping @siying @ajkr

**Comment by siying** [comment]
Approved. Please merge after all tests pass.

**Pull request closed without merging**